src/shared/generator.c

```diff
@@ -736,12 +736,16 @@ int generator_write_veritysetup_service_section(
         return 0;
 }
 
-void log_setup_generator(void) {
-        /* Disable talking to syslog/journal (i.e. the two IPC-based loggers) if we run in system context. */
-        if (cg_pid_get_owner_uid(0, NULL) == -ENXIO /* not running in a per-user slice */)
-                log_set_prohibit_ipc(true);
+void log_setup_generator(bool normal_invocation) {
+        if (normal_invocation) {
```

Comment:
It may be overkill, but how about to use the similar logic as `should_parse_proc_cmdline()` in log.c, instead of `argc == 4`?